ssh/tailssh/tailssh.go

```diff
@@ -122,19 +125,37 @@ func (srv *server) newSSHServer() (*ssh.Server, error) {
 		},
 		Version:                     "SSH-2.0-Tailscale",
 		LocalPortForwardingCallback: srv.mayForwardLocalPortTo,
-		NoClientAuthCallback: func(m gossh.ConnMetadata) (*gossh.Permissions, error) {
-			if srv.requiresPubKey(m.User(), toIPPort(m.LocalAddr()), toIPPort(m.RemoteAddr())) {
-				return nil, errors.New("public key required") // any non-nil error will do
+		NoClientAuthCallback: func(conn gossh.ConnMetadata) (*gossh.Permissions, error) {
```

Comment:
upstream gliderlabs/ssh doesn't have this hook. I had removed it in my https://github.com/tailscale/tailscale/pull/4441 and moved it to ServerConfigCallback instead. Since you're already using ServerConfigCallback, do you want to also do that in this PR, then also removing that modification to our gliderlabs/ssh fork in this PR too like #4441?